### ISSUE TITLE
Every call to to_param results in an extra network call.

### ISSUE THREAD

**Issue by iHiD** [neo4jrb/neo4j#826]
Hello. Thanks for all your fantastic work in making this!

tl;dr; An index page of results is resulting in 150 extra network calls.

I'm only just starting to play with this so I may be missing something. 
- On an index page, I have a list of 50 items. 
- Each has the standard view/edit/delete links. 
- I noticed that the page loading was very slow (1s).
- Commenting out the links sped up dramatically (100ms)
- I tried stubbing a to_param in the model and it sped back up again.
- I stepped through the executing code and discovered that it's making a network call per lookup in the  [to_key](https://github.com/neo4jrb/neo4j/blob/41486a864cd42811bf283e9060c66a3a1cc1d6fd/lib/neo4j/shared/identity.rb#L11) method calling out to `exist?` in the [cypher_node.rb](https://github.com/neo4jrb/neo4j-core/blob/master/lib/neo4j-server/cypher_node.rb#L141).

I'm guessing this codepath is there to check the record has at some point been persisted and so has a key, and that the check to see if it **still** exists is inadvertent.

Would you except a PR to match ActiveRecord's behaviour here and not check whether it's still persisted, or is this behaviour intentional?

Thanks again! :blue_heart:

**Comment by subvertallchris** [comment]
Really great catch! I've been trying to weed these out but clearly missed this one. If you change that `persisted?` to `_persisted_obj`, you'll fix it. Totally down to take a PR if you'd like to do it, otherwise I can get it in there at some point today.

**Comment by iHiD** [comment]
I'll knock it together now. Thanks :)
